interfaces/repo_test.go

```diff
@@ -790,12 +790,77 @@ func (s *RepositorySuite) TestPlugConnectionsReturnsCorrectData(c *C) {
 	c.Assert(err, IsNil)
 	users := s.testRepo.PlugConnections(s.plug.Snap, s.plug.Name)
 	c.Assert(users, DeepEquals, []*Slot{s.slot})
-	// After revoking the result is empty again
+	// After disconnecting the result is empty again
 	err = s.testRepo.Disconnect(s.plug.Snap, s.plug.Name, s.slot.Snap, s.slot.Name)
 	c.Assert(err, IsNil)
 	c.Assert(s.testRepo.PlugConnections(s.plug.Snap, s.plug.Name), HasLen, 0)
 }
 
+// Tests for Repository.Interfaces()
+
+func (s *RepositorySuite) TestInterfacesSmokeTest(c *C) {
+	err := s.testRepo.AddPlug(s.plug)
+	c.Assert(err, IsNil)
+	err = s.testRepo.AddSlot(s.slot)
+	c.Assert(err, IsNil)
+	// After connecting the result is as expected
+	err = s.testRepo.Connect(s.plug.Snap, s.plug.Name, s.slot.Snap, s.slot.Name)
+	c.Assert(err, IsNil)
+	ifaces := s.testRepo.Interfaces()
+	c.Assert(ifaces, DeepEquals, &Interfaces{
+		Plugs: []*Plug{
+			&Plug{
+				Name:      s.plug.Name,
+				Snap:      s.plug.Snap,
+				Interface: s.plug.Interface,
+				Label:     s.plug.Label,
+				Connections: []SlotRef{
+					{
+						Snap: s.slot.Snap,
+						Name: s.slot.Name,
+					},
+				},
+			},
+		},
+		Slots: []*Slot{
+			&Slot{
+				Snap:      s.slot.Snap,
+				Name:      s.slot.Name,
+				Interface: s.slot.Interface,
+				Label:     s.slot.Label,
+				Connections: []PlugRef{
+					{
+						Snap: s.plug.Snap,
+						Name: s.plug.Name,
+					},
+				},
+			},
+		},
```

Comment:
Please use the compact representation I provided as an example in the follow up branches. There are about 10 lines too many here.